File: search/binary-search/binary-search-recursive.py
import math

arr = [1, 2, 3, 4, 5]
searchNo = 5

def binary_search(arr, searchNo, low, high):
    if low > high:
        return -1

    mid = math.floor((low + high) / 2)

    if arr[mid] == searchNo:
        return mid

    if arr[mid] < searchNo:
        low = mid + 1
        return binary_search(arr, searchNo, low, high)
    else:
        high = mid - 1
        return binary_search(arr, searchNo, low, high)

print(binary_search(arr, searchNo, 0, len(arr) - 1))

File: search/binary-search/binary-search-recursive.js
const arr = [1, 2, 3, 4, 5];
const searchNo = 5;

function binarySearch(arr, x, low, high) {

    // Base Condition
    if (low > high) return -1;

    // Find the middle index
    let mid = Math.floor((low + high) / 2);

    // Compare mid with given key x
    if (arr[mid] === x) return mid;

    // If element at mid is greater than x,
    // search in the left half of mid
    if(arr[mid] > x) {
        return binarySearch(arr, x, low, mid - 1);
    } else {

        // If element at mid is smaller than x,
        // search in the right half of mid
        return binarySearch(arr, x, mid + 1, high);
    }
}

console.log(binarySearch(arr, searchNo, 0, arr.length));